app/src/main/java/de/danoeh/antennapod/fragment/swipeactions/SwipeActions.java

```diff
@@ -0,0 +1,282 @@
+            Arrays.asList(new AddToQueueSwipeAction(), new MarkUnplayedSwipeAction(),
+                    new StartDownloadSwipeAction(), new MarkFavouriteSwipeAction(),
+                    new MarkPlayedSwipeAction(), new RemoveFromQueueSwipeAction())
+    );
+
+    private RecyclerView recyclerView;
+    public ItemTouchHelper itemTouchHelper;
+    private final Fragment fragment;
+    private final String tag;
+    private FeedItemFilter filter = null;
+
+    private NewSwipeCallback newSwipeCallback = SimpleSwipeCallback::new;
+
+    public SwipeActions(Fragment fragment, String tag) {
+        this.fragment = fragment;
+        this.tag = tag;
+        itemTouchHelper();
+    }
+
+    public void setFilter(FeedItemFilter filter) {
+        this.filter = filter;
+    }
+
+    public SwipeActions attachTo(RecyclerView recyclerView) {
+        this.recyclerView = recyclerView;
+        itemTouchHelper.attachToRecyclerView(recyclerView);
+        //reload as settings might have changed
+        fragment.getLifecycle().addObserver((LifecycleEventObserver) (source, event) -> {
+            if (event == Lifecycle.Event.ON_RESUME) {
+                resetItemTouchHelper();
+            }
+        });
+
+        return this;
+    }
+
+    public void itemTouchHelper() {
+        itemTouchHelper = new ItemTouchHelper(newSwipeCallback.construct());
+    }
+
+    private static List<SwipeAction> getPrefs(Context context, String tag, String defaultActions) {
+        SharedPreferences prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
+        String prefsString = prefs.getString(PREF_SWIPEACTIONS + tag, defaultActions);
+
+        ArrayList<SwipeAction> actions = new ArrayList<>();
+
+        if (!prefsString.isEmpty()) {
+            String[] rightleft = prefsString.split(",");
+
+            for (String s : rightleft) {
+                actions.add(Stream.of(swipeActions).filter(a -> a.id().equals(s)).single());
+            }
+        }
+
+        return actions;
+    }
+
+    private static List<SwipeAction> getPrefs(Context context, String tag) {
+        return getPrefs(context, tag, "");
+    }
+
+    public static List<SwipeAction> getPrefsWithDefaults(Context context, String tag) {
+        String defaultActions;
+        switch (tag) {
+            /*case InboxFragment.TAG:
+                defaultActions = new int[] {ADD_TO_QUEUE, MARK_UNPLAYED};
+                break;*/
+            case QueueFragment.TAG:
+                defaultActions = SwipeAction.REMOVE_FROM_QUEUE + "," + SwipeAction.REMOVE_FROM_QUEUE;
+                break;
+            default:
+            case EpisodesFragment.TAG:
+                defaultActions = SwipeAction.MARK_FAV + "," + SwipeAction.START_DOWNLOAD;
+                break;
+        }
+
+        return getPrefs(context, tag, defaultActions);
+    }
+
+    public static Boolean getNoActionPref(Context context, String tag) {
```

Comment:
No boxing needed, I think

```suggestion
    public static boolean getNoActionPref(Context context, String tag) {
```

Also, the name is a bit unclear to me. If it does what I think it does, `isSwipeActionEnabled` would be more clear.